src/docs/arch-checkpoint.dox

```diff
@@ -19,46 +21,71 @@ WT_CONNECTION::rollback_to_stable.
+the \c checkpoint_lock. This lock ensures only one checkpoint takes place at time. More granular locks
+such as, dhandle, schema and flush locks are taken throughout the checkpoint process.
+
+## Eviction ##
+
+Before the checkpoint transaction begins (pinning content), the @ref arch-eviction "eviction server" is leveraged to
+reduce the dirty content in the cache. This has the benefit of utilizing the multi-threaded
+functionality of eviction and reduces the work required during the critical section of the
+checkpoint. The thread performing the checkpoint operation will wait for eviction to
+reach the target dirty content ratio unless no progress is made..
+
+## Prepare ##
 
 Checkpoint prepare sets up the checkpoint, it begins the checkpoint transaction, updates the global
 checkpoint state and gathers a list of handles to be checkpointed. A global schema lock wraps
 checkpoint prepare to avoid any tables being created or dropped during this phase, additionally the
 global transaction lock is taken during this process as it must modify the global transaction state,
 and to ensure the \c stable_timestamp doesn't move ahead of the snapshot taken by the checkpoint
-transaction. Each handle gathered refers to a specific b-tree. The set of b-trees gathered by the
-checkpoint varies based off configuration. Additionally clean b-trees, i.e. b-trees without any
-modifications are excluded from the list, with an exception for specific checkpoint configuration
-scenarios.
+transaction. The connection dhandle list is walked one handle at a time where each handle gathered
+refers to a specific b-tree. The set of b-trees gathered by the checkpoint varies based off
+configuration. Additionally clean b-trees, i.e. b-trees without any modifications are excluded from
+the list, with an exception for specific checkpoint configuration scenarios. See
+\ref skipping-checkpoints "Skipping checkpoints" for scenarios where b-trees are skipped during
+checkpoint. B-trees included in the checkpoint list then have all earlier checkpoints marked for
+deletion. Previous checkpoints can only be discarded if the associated blocks are no longer used.
 
-_The data files checkpoint:_
+## Data files checkpoint ##
 
 Data files in this instance refer to all the user created files. The main work of checkpoint is done
 here, the array of b-tree's collected in the prepare stage are iterated over. For each b-tree, the
 tree is walked and all the dirty pages are reconciled. Clean pages are skipped to avoid unnecessary
 work. Pages made clean ahead of the checkpoint by eviction are still skipped regardless of whether
 the update written by eviction is visible to the checkpoint transaction. The checkpoint guarantees
-that a clean version of every page in the tree exists and can be written to disk.
+that a clean version of every page in the tree exists and can be written to disk. The
+@ref arch-block "Block Manager" is then responsible for managing the extent lists modified by the
```

Comment:
Again "Block Manager" should be: the block manager subsystem (no double quotes and use subsystem)

Reply:
Per your earlier comment "block manager subsystem"